### PR TITLE
add s1.meta.azimuth_fmrate

### PR DESCRIPTION
**Pull request by agrouaze**
address issue #52 

- [x] add mappings
- [x] add compound var
- [x] add s1meta property
- [x] validation

### CODE REVIEW

**Comment by agrouaze** [comment]
image

**Comment by agrouaze** [comment]
image

File: src/xsar/sentinel1_meta.py [hunk 1]
```diff
@@ -625,6 +625,12 @@ def image(self):
         img_dict['history'] = self.xml_parser.get_compound_var(self.files['annotation'].iloc[0], 'image', describe=True)
         return img_dict
 
+    @property
+    def azimuth_fmrate(self):
+        fmrates = self.xml_parser.get_compound_var(self.files['annotation'].iloc[0], 'azimuth_fmrate')
+        fmrates['history'] = self.xml_parser.get_compound_var(self.files['annotation'].iloc[0], 'azimuth_fmrate', describe=True)
```
Comment (oarcher) [inline src/xsar/sentinel1_meta.py:631]: if `fmrates` is an xarray dataset, history should be in `fmrates.attrs['history']`, not in `fmrates['history']`

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_meta.py (hunk 1 above)
Comment (agrouaze) [reply src/xsar/sentinel1_meta.py]: Correction done!
image

File: src/xsar/sentinel1_xml_mappings.py [hunk 2]
```diff
@@ -31,6 +32,7 @@
 datetime64_array = lambda x: np.array([np.datetime64(date_converter([sx])) for sx in x])
 int_1Darray_from_string = lambda x: np.fromstring(x[0], dtype=int, sep=' ')
 float_2Darray_from_string_list = lambda x: np.vstack([np.fromstring(e, dtype=float, sep=' ') for e in x])
+float_list_of_list_from_string = lambda x: [np.fromstring(e, dtype=float, sep=' ') for e in x] #usefull in case of count that can change from one element to the other
```
Comment (oarcher) [inline src/xsar/sentinel1_xml_mappings.py:35]: see #53 : https://github.com/umr-lops/xsar/pull/51/files#r822099199 
( use `list_of_float_1D_array_from_string` )

File: src/xsar/sentinel1_xml_mappings.py [hunk 3]
```diff
@@ -408,6 +419,32 @@ def orbit(time, frame, pos_x, pos_y, pos_z, vel_x, vel_y, vel_z,orbit_pass,platf
...
+    azimuthtime
+    t0
+    c0
+    c1
+    c2
+    polynomial
+
+    Returns
+    -------
+
```
Comment (oarcher) [inline src/xsar/sentinel1_xml_mappings.py:436]: describe the returned dataset

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_xml_mappings.py (hunk 3 above)
Comment (agrouaze) [reply src/xsar/sentinel1_xml_mappings.py]: fixed.

File: src/xsar/sentinel1_xml_mappings.py (hunk 2 above)
Comment (agrouaze) [reply src/xsar/sentinel1_xml_mappings.py]: fixed

File: src/xsar/sentinel1_meta.py [hunk 4]
```diff
@@ -663,6 +663,12 @@ def image(self):
         img_dict['history'] = self.xml_parser.get_compound_var(self.files['annotation'].iloc[0], 'image', describe=True)
         return img_dict
 
+    @property
+    def azimuth_fmrate(self):
```
Comment (oarcher) [inline src/xsar/sentinel1_meta.py:667]: This property has to be documented in the user manual, so it need a docstring

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_meta.py (hunk 4 above)
Comment (agrouaze) [reply src/xsar/sentinel1_meta.py:667]: doc string added.

**Pull request merged**